platform/security/filter/security-filter-csrf/src/main/java/org/codice/ddf/security/filter/csrf/CsrfFilter.java

```diff
@@ -114,23 +139,29 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
   }
 
   /**
-   * Returns true if both URLs have the same host and port. If either are null or empty, false is
-   * returned
+   * Returns true if the supplied URL has an authority (hostname:port) that matches one of the
+   * system's trusted authorities. If the URL is blank or malformed, false is returned.
    *
    * @param source source URL
-   * @param target destination URL
    * @return true if matching, false if different or a parsing error occurs
    */
-  private Boolean isSameOrigin(String source, String target) {
-    if (StringUtils.isBlank(source) || StringUtils.isBlank(target)) {
+  private Boolean isTrustedAuthority(String source) {
+    if (StringUtils.isBlank(source)) {
       return false;
     } else {
       try {
-        String sourceOrigin = new URL(source).getAuthority();
-        String targetOrigin = new URL(target).getAuthority();
-        return (sourceOrigin.equals(targetOrigin));
+        URL url = new URL(source);
+        String sourceAuthority;
+        // if no port os specified, assume default http/https ports
```

Comment:
did you mean `is`??????????????????

Reply:
[yes :(](https://media.giphy.com/media/3o7WTqo27pLRYxRtg4/giphy.gif)

Reply:
![](https://media.giphy.com/media/3o7WTqo27pLRYxRtg4/giphy.gif)